internal/api/describe_test.go

```diff
@@ -0,0 +1,74 @@
+package api
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestDescribeFull(test *testing.T) {
+	path, err := common.GetAPIDescriptionFilepath()
+	if err != nil {
+		test.Fatalf("Unable to get the API description filepath: '%v'.", err)
+	}
+
+	var expectedDescription core.APIDescription
+	err = util.JSONFromFile(path, &expectedDescription)
+	if err != nil {
+		test.Fatalf("Failed to load api.json: '%v'.", err)
+	}
+
+	fullDescription := Describe(*GetRoutes())
+
+	if !reflect.DeepEqual(&expectedDescription, fullDescription) {
+		test.Fatalf("Unexpected API Descriptions. Expected: '%s', actual: '%s'.",
+			util.MustToJSONIndent(expectedDescription), util.MustToJSONIndent(fullDescription))
+	}
+}
+
+func TestDescribeEmptyRoutes(test *testing.T) {
+	routes := []core.Route{}
+	description := Describe(routes)
+
+	if len(description.Endpoints) != 0 {
+		test.Errorf("Unexpected number of endpoints. Expected: '0', actual: '%d'.", len(description.Endpoints))
+	}
+}
+
+func TestDescribeDuplicateBasePaths(test *testing.T) {
```

Comment:
My previous comment still stands here.
Duplicate paths for routes are allowed (since they can have a different HTTP method).

Also, this test seems off to me.
Who decided the semantics that the second path should survive?

I think there are two good options here:
1) Remove this test and don't encourage people to have duplicate API endpoints.
2) Formalize the functionality for duplicate routes and check for them.